### PR TITLE
Check for GNU Make before trying to build

### PR DESCRIPTION
**Pull request by ararslan**
Would have caught #44.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -1,8 +1,17 @@
-if Sys.KERNEL === :Darwin && (!success(`command -v xcode-select`) || isempty(readchomp(`xcode-select -p`)))
+using Compat
```
Comment (tkelman) [inline deps/build.jl]: needs to be in REQUIRE

Comment (ararslan) [reply deps/build.jl]: Ah, good catch. I just assumed we already had a Compat dependency.

File: deps/build.jl [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+using Compat.Sys: isapple, iswindows
+
+if isapple() && (!success(`command -v xcode-select`) || isempty(readchomp(`xcode-select -p`)))
     error("Building Rmath on macOS requires the Xcode command line tools to be installed.\n",
           "You can install them from the command line using `xcode-select --install`.")
 end
 
+if !iswindows() && !(success(`command -v make`) || success(`command -v gmake`))
+    error("Building Rmath requires GNU Make to be installed. You can install it from your\n",
```
Comment (simonbyrne) [inline deps/build.jl]: I don't think `command` is available on non-BSD.

Comment (simonbyrne) [reply deps/build.jl]: could just try:
```
success(`make -v`)
```

Comment (ararslan) [reply deps/build.jl]: `command` is POSIX, but I guess it needs to be ``success(`sh -c "command -v make"`)`` to be fully correct. Won't ``success(`$(BinDeps.MAKE_CMD) -v`)`` fail with `ENOENT` if Make isn't installed?

**Comment by andreasnoack** [comment]
No longer relevant after #55

**Pull request closed without merging**